seqlike/SeqLike.py

```diff
@@ -606,7 +608,7 @@ def __getitem__(self, index) -> "SeqLike":
                 elif self._type == "NT":
                     _nt_record = deepcopy(self._nt_record)[idx]
                     try:
-                        _aa_record = _nt_record.translate()
+                        _aa_record = _nt_record.translate(gap=gap_letter)
```

Comment:
This change is consistent with https://github.com/modernatx/seqlike/blob/main/seqlike/SeqLike.py#L291

Reply:
Without this, the following unittest fails because SeqLike tries to back translate "---" as a codon:
```
FAILED tests/test_SeqLike.py::test_slice_with_back_translation[TTC---GACACTGCA-nt] - Bio.Data.CodonTable.TranslationError: Codon '---' is invalid
```